cohortextractor2/validate_dummy_data.py

```diff
@@ -102,9 +102,18 @@ def date_validator(value):
         except ValueError:
             datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
 
-    def category_validator(value, categories):
-        category_type = type(list(categories)[0])
-        if category_type(value) not in categories:
+    def category_validator(value, categories, default_category):
+        """Ensure that a category value is one of the expected categories, or the default"""
+        # The default can be a different type (e.g. None, or the string "missing")
+        category_type = type(categories[0])
+        try:
+            value = category_type(value)
+        except ValueError:
```

Comment:
I think this can raise `TypeError` as well as `ValueError` e.g. `int(object())` is a `TypeError`

Reply:
Didn't think of that - yes, will catch both